### DISCUSSION TITLE
씬 이동 후 다시 이동 전 씬으로 돌아왔을 때 같은 룸으로 접속이 되는지 궁금합니다.

### DISCUSSION THREAD

**Discussion by LeeYongSu1** [naverz/zepeto-studio-kor#701] [5. Q&A]
World Ver: 1.3.0

App Ver: 3.12.1

구현하려는 기능:
안녕하세요 월드 내 이동을 씬 이동으로 구현 하려고 하고 있습니다.

문제점
1. 제공해 주신 예제에서 ClientStart와 Multiplay 오브젝트를 Dontdestroyonload해 다른 씬으로 이동 후 다시 이동 전 씬으로 돌아오면
데이터를 보존해 씬 이동이 가능하고, 서버와 통신도 되는데 룸 접속자 목록에 유저들이 나타나지 않습니다.

2. 서버와 연결하는 Multiplay오브젝트는 Dontdestroyonload하지 않고 씬 이동 후 다시 돌아오면 RoomJoin이벤트가 발생해 다시 룸에 접속하게 되어 룸 접속자 목록에는 유저들이 나타납니다만, 다시 같은 룸에 접속을 하게 되는 것인지 궁금합니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 말씀해주신 내용에 대해 개선작업이 진행 중입니다.
반영 시점은 조율 중이며 확정되면 공유 드리겠습니다.
감사합니다.

**Comment by YSHFM** [comment]
안녕하세요,
https://www.mettit.com/pt/view/PRE20220523142110737
해당 링크 참고 하시면 도움 되실 것 같습니다~

**Reply by YSHFM** [reply]
저도 동일한 문제로 인스턴스가 되지 않거나 , GetPlayer를 할 수 없거나, 2개가 인스턴스가 되거나 하는 이슈가 여러번 있었습니다만,
씬 이동 시에 생성 삭제하는 방법으로 구현에 성공했습니다.
아무래도 비효율적일 수 밖에 없는데, api에 개선이 필요해 보입니다.
